src/api/VoteMonitor.Api.Location/Handlers/PollingStationHandler.cs

```diff
@@ -24,48 +26,49 @@ public PollingStationHandler(VoteMonitorContext context, IMapper mapper, ILogger
 
         protected override async Task<int> HandleCore(PollingStationCommand request)
         {
-            var records = request.PollingStationsDTOs;
-            Random random = new Random();
+            var random = new Random();
+
             try
             {
                 //import the new entities
-                using(var transaction = await _context.Database.BeginTransactionAsync())
+                using (var transaction = await _context.Database.BeginTransactionAsync())
                 {
-                    int id = 0;
-                    foreach(var record in records)
+                    var id = 100;
+                    var newPollingStations = new List<PollingStation>();
+                    var counties = _context.Counties.ToDictionary(c => c.Code, c => c.Id);
+
+                    foreach (var record in request.PollingStationsDTOs)
                     {
-                        PollingStation pollingStation = _mapper.Map<PollingStation>(record);
+                        var pollingStation = _mapper.Map<PollingStation>(record);
                         pollingStation.Id = id++;
-                        County county = _context.Counties
-                                            .Where(c => c.Code == record.CodJudet)
-                                            .First();
-                        pollingStation.IdCounty = county.Id;
+                        pollingStation.IdCounty = counties[record.CodJudet];//county.Id;
                         pollingStation.Coordinates = null;
-                        pollingStation.County = county;
                         pollingStation.TerritoryCode = random.Next(10000).ToString();
-                        _context.PollingStations.Add(pollingStation);
+
+                        newPollingStations.Add(pollingStation);
```

Comment:
Added the polling station collection in a local list, not attached to the context